src/buffer.rs

```diff
@@ -441,9 +442,7 @@ impl<T: Element> PyBuffer<T> {
 
     fn copy_to_slice_impl(&self, py: Python, target: &mut [T], fort: u8) -> PyResult<()> {
         if mem::size_of_val(target) != self.len_bytes() {
-            return Err(exceptions::PyBufferError::new_err(
-                "Slice length does not match buffer length.",
-            ));
+            return Err(invalid_sequence_length(self.item_count(), target.len()));
```

Comment:
I'm sorry for communicating poorly earlier; the two uses of `invalid_sequence_length` were actually correct to stay as `BufferError`.

I'll push a separate PR to improve these error messages; in this PR can you please just reset them to what they were (ultimately leaving `buffer.rs` unchanged by this PR, to avoid a merge conflict).